pkg/trace/writer/trace.go

```diff
@@ -32,23 +33,23 @@ const pathTraces = "/api/v0.2/traces"
 // a flush is triggered; replaced in tests.
 var MaxPayloadSize = 3200000 // 3.2MB is the maximum allowed by the Datadog API
 
-// SampledSpans represents the result of a trace sampling operation.
-type SampledSpans struct {
-	// Trace will contain a trace if it was sampled or be empty if it wasn't.
-	Traces []*pb.APITrace
-	// Events contains all APM events extracted from a trace. If no events were extracted, it will be empty.
-	Events []*pb.Span
+// SampledChunks represents the result of a trace sampling operation.
+type SampledChunks struct {
+	// TracerPayload contains all the chunks that were sampled as part of processing a payload.
+	TracerPayload *pb.TracerPayload
 	// Size represents the approximated message size in bytes.
 	Size int
-	// SpanCount specifies the total number of spans found in Traces.
+	// SpanCount specifies the number of spans that were sampled as part of a trace inside the TracerPayload.
```

Comment:
Should we mention that it excludes events? I think we should make that clear.
```suggestion
	// SpanCount specifies the number of spans that were sampled as part of a trace inside the TracerPayload.
	// SpanCount does not count spans included as APM events.
```

Reply:
> SpanCount does not count spans included as APM events.

Actually it does, if the trace was kept.

Before:
* `SpanCount` - the number of spans in `Traces`
* `EventCount` - the number of spans in `Transactions`

After:
* `SpanCount` - the number of spans in non dropped `Chunks`
* `EventCount` - the number of spans flagged `_dd.analyzed:1` in `Chunks`

Because we removed `Transactions`, we need to double count some spans in order to reproduce the same stats.

Reply:
I think this creates very big confusion. You are currently only incrementing the `SpanCount` field if the trace is not dropped (`!chunk.DroppedTrace`), in which case `SpanCount` is counting both event and non-event spans. When the trace _is_ dropped, you are not counting these spans in `SpanCount`, which makes no sense. This is only used for metrics, and I think we should count _all_ spans regardless of whether they are events or not. As a result, I'm proposing that you _always_ count the spans inside the `SampledChunks`, regardless of whether `chunk.DroppedTrace` is true or not.

Can we make that change? We want to measure (in the metric) all spans going through. Doesn't matter if they are events or not. If anyone wants to see how many non-event spans there are, they can always do `datadog.trace_agent.trace_writer.spans` - `datadog.trace_agent.trace_writer.events`.

Reply:
You're right. Ignore this comment. Resolving.